nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -199,13 +199,13 @@ public FileSystemRepository(final NiFiProperties nifiProperties) throws IOExcept
             for (final Map.Entry<String, Path> container : containers.entrySet()) {
                 final String containerName = container.getKey();
 
-                final long capacity = Files.getFileStore(container.getValue()).getTotalSpace();
+                final long capacity = container.getValue().toFile().getTotalSpace();
```

Comment:
If there is a problem, FileStore.getTotalSpace() throws IOException but File.getTotalSpace() returns 0.  I think you should throw an IOException if capacity == 0, in order to maintain similar behavior here.